### PR TITLE
[bugfix] Fix a bug which results in incorrect reporting of violations

### PR DESCRIPTION
**Pull request by jzuhone**
For the details of why this PR was created I defer to the description at the corresponding PR acisops/acis_thermal_check#18.

### CODE REVIEW

**Comment by Gregg140** [comment]
If I remember correctly, get_last_data_date() was to find the last available telemetry time. If so you can eliminate that by fetching the telemetry with no stop date  and noting the last date you get.

**Review by Gregg140 — APPROVED** [review]
Look good to me

**Review by richardjedgar — APPROVED** [review]
Looks OK to me.

**Comment by jeanconn** [comment]
I can confirm it looks to me like I in this branch I can now import acisfp_check successfully even if I don't have an available eng_archive (which is a *good* thing.  especially for scenarios when I'm building a clean ska environment).

**Review by Gregg140 — APPROVED** [review]
Looks fine to me.

**Comment by jzuhone** [comment]
@taldcroft did you have a chance to look at this and the corresponding PR acisops/acis_thermal_check#18? It turns out this PR fixes the reported violation in this week's load review for the FP temperature, which is not a real violation: 

http://cxc.cfa.harvard.edu/acis/FP_thermPredic/MAY2118/oflsa/

File: acisfp_check/__init__.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-__version__ = "2.2.0"
+__version__ = "2.2.1"
```
Comment (taldcroft) [inline acisfp_check/__init__.py]: Being pedantic, this PR changes the API so should go to version "2.3".  (By making `search_obsids_for_viols` into a method and changing the call signature).

**Review by taldcroft — APPROVED** [review]
With the testing described in https://github.com/acisops/acis_thermal_check/pull/18 this looks fine to me (modulo the nitpick which is not required to fix).

**Review by jeanconn — APPROVED** [review]

**Review by taldcroft — APPROVED** [review]

**Pull request merged**